tests/v2/validation/configmaps/generated_fields_test.go

```diff
@@ -0,0 +1,174 @@
+	"strings"
+	"testing"
+)
+
+const (
+	APIVersion = "v1"
+	Kind       = "ConfigMap"
+)
+
+type ConfigMapTestSuite struct {
+	suite.Suite
+	client           *rancher.Client
+	steveClient      *steveV1.Client
+	session          *session.Session
+	cluster          *management.Cluster
+	httpClient       *http.Client
+	configMapPayload *configMapPayload
+}
+
+type configMapPayload struct {
+	APIVersion string            `yaml:"apiVersion"`
+	Data       map[string]string `yaml:"data"`
+	Kind       string            `yaml:"kind"`
+	Metadata   struct {
+		Fields          string            `yaml:"fields"`
+		Relationships   string            `yaml:"relationships"`
+		State           string            `yaml:"state"`
+		Labels          map[string]string `yaml:"labels"`
+		Name            string            `yaml:"name"`
+		Namespace       string            `yaml:"namespace"`
+		ResourceVersion string            `yaml:"resourceVersion"`
+		UID             string            `yaml:"uid"`
+	} `yaml:"metadata"`
+}
+
+func (c *ConfigMapTestSuite) TearDownSuite() {
+	c.session.Cleanup()
+}
+
+func (c *ConfigMapTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	c.session = testSession
+	client, err := rancher.NewClient("", c.session)
+	require.NoError(c.T(), err)
+	c.client = client
+	c.steveClient = client.Steve
+	c.httpClient = &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+	}
+}
+
+func (c *ConfigMapTestSuite) TestSteveGeneratedFields() {
+
+	configData := map[string]string{
+		"env": "qa",
+	}
+	labels, annotations := map[string]string{}, map[string]string{}
+	configMapName := namegenerator.AppendRandomString("test-configmap")
+	configmap, err := configmaps.CreateConfigMap(c.client, c.client.RancherConfig.ClusterName, configMapName, "auto-generated", "default", configData, labels, annotations)
+	require.NoError(c.T(), err)
+
+	logrus.Infof("created a configmap(%v, configmap).............", configmap.Name)
+
+	c.configMapPayload = &configMapPayload{}
+	copyConfigMapToPayload(configmap, c.configMapPayload)
+
+	// Update the configmap payload with the fields that we want to test.
+	c.configMapPayload.Metadata.Fields = "test-fields"
+	c.configMapPayload.Metadata.Relationships = "test-relationships"
+	c.configMapPayload.Metadata.State = "test-state"
+	c.configMapPayload.Data["foo"] = "bar"
+
+	headers, _, err := postAction(c, configmap.Namespace, configmap.Name)
+	require.NoError(c.T(), err)
+
+	// Check for warnings in the response headers.
+	warnings, ok := headers["Warning"]
+	var failWarnings []string
+	if ok {
+		for _, warning := range warnings {
+			// If the warning message starts with "299", then log it and add it to failWarnings.
+			if strings.HasPrefix(warning, "299") {
+				logrus.Printf("Warning header found: %s", warning)
+				failWarnings = append(failWarnings, warning)
+			}
+		}
+	}
+
+	// If there were any "299" warnings, fail the test.
+	if len(failWarnings) > 0 {
```

Comment:
qq: we should not be getting a single warning and prining any/all of the warning is it ?

Reply:
Yeah there shouldn't be any warnings. And if there are we log them and fail